# Stellar Atmospheres 

## The Description of the Radiation Field
Astronomers receive light only from the outermost layer of the star's atmosphere that overly an opaque interior.  The *temperature*, *density*, and *composition* of this layer determine the features of the star's spectrum, where we must describe how light travels through the gas to interpret the observed spectral lines properly.

### The Specific and Mean Intensities
Recall the [Planck function](https://saturnaxis.github.io/ModernAstro/Chapter_3/continuous-spectrum-of-light.html#the-planck-function-and-astrophysics) was based upon the wavelength-dependent flux of blackbody radiation $B_\lambda(T)$ from an element of surface area $dA$.  A similar procedure can be used to define the intensity of a cone of light.  The light is measured in spherical coordinates by the angle $\theta$, which is the direction perpendicular to the surface $dA$ so that $dA\cos\theta$ is an area projected onto a plane perpendicular to the propagation direction of the light.  The light energy is wavelength dependent ($E = hc/\lambda$) and thus can be defined as

$$ E_\lambda \equiv \frac{\partial E}{\partial \lambda}, $$

where $E_\lambda d\lambda$ is the energy carried by the rays in the light cone in a time interval $dt$.  The **specific intensity** of the rays is defined as

\begin{align}  I_\lambda \equiv \frac{\partial I}{\partial \lambda} \equiv \frac{E_\lambda d\lambda}{d\lambda\:dt\:dA \cos\theta\:d\Omega}, \end{align} 

which carries the units of energy per time per Area per steradian *sr* ([solid angle)](https://en.wikipedia.org/wiki/Steradian) or more simply, energy per everything.

```{figure-md} intensity-fig
<img src="Figure_1.jpg" alt="Intensity"  width="500px">

Schematic for the intensity $I_\lambda$ that passes through a surface. Image credit: Carroll & Ostlie (2007).
```

```{note}
There is a $d\lambda$ factor in the numerator and denominator, which is a placeholder for the alternate definition of specific intensity in terms of frequency $I_\nu$.  Recall from the [Planck function](https://saturnaxis.github.io/ModernAstro/Chapter_3/continuous-spectrum-of-light.html#the-planck-function-and-astrophysics) using $d\nu = -\frac{c}{\lambda^2}d\lambda$ as a conversion factor.
```

The specific intensity is usually referred to simply as the *intensity* and in spherical coordinates,

```{math}
:label: specific_intensity
E_\lambda d\lambda = I_\lambda d\lambda\:dt\:dA \cos\theta\:d\Omega = I_\lambda d\lambda\:dt\:dA \cos\theta \sin\theta d\theta\:d\phi,
```

with the substitution for the solid angle $d\Omega = \sin\theta d\theta \:d\phi$.  The Planck function $B_\lambda$ is an example of the specific intensity for the special case of blackbody radiation.  In general, the light energy is not required to vary with wavelength (e.g., monochromatic sources or lasers).

The energy of a single light ray *does not spread out (or diverge)*, however the specific intensity $I_\lambda$ is typically used to describe a bundle of light rays and thus varies with direction.  The **mean intensity** of the radiation is determined by integrating over all directions and normalizing the result by $4\pi\:{\rm sr}$ (i.e., the solid angle enclosed by a sphere) to obtain $\langle I_\lambda \rangle$, the average value of $I_\lambda$.  In spherical coordinates, this average value is

```{math}
:label: mean_intensity
\langle I_\lambda \rangle \equiv \frac{1}{4\pi} \int_{\phi = 0}^{2\pi} \int_{\theta=0}^{\pi} I_\lambda \sin\theta d\theta\:d\phi.
```

For an isotropic radiation field (i.e., one with the same intensity in all directions), the mean intensity is the intensity and since blackbody radiation is isotropic, then $\langle I_\lambda \rangle = B_\lambda$.

### The Specific Energy Density
Consider a small cylinder (open at both ends) with perfectly reflecting walls inside, where light enters the base of the cylinder and bounces until it exits through the top of the cylinder.  The radiation that enters the cylinder at an angle $\theta$ travels through the cylinder in a time $dt = dL/(c\cos\theta)$ and the energy inside (within a wavelength range $\lambda$ and $\lambda + d\lambda$) is

$$ E_\lambda d\lambda = I_\lambda d\lambda\:dt\:dA \cos\theta\:d\Omega = I_\lambda d\lambda\:dA\:d\Omega\:\frac{dL}{c}, $$

using Eqn. {eq}`specific_intensity` and the substitution of $dt$.  The quantity $dA dL$ is the volume of the cylinder, where the **specific energy density** ($u_\lambda d\lambda$) is found by normalizing the energy $E_\lambda d\lambda$ by the volume of the cylinder, integrating over all the solid angles, and using Eqn. {eq}`mean_intensity`:

\begin{align} 
u_\lambda d\lambda = \frac{1}{c} \int_{\phi = 0}^{2\pi} \int_{\theta=0}^{\pi} I_\lambda d\lambda\:\sin\theta d\theta\:d\phi = \frac{4\pi}{c} \langle I_\lambda \rangle d\lambda. 
\end{align}

```{figure-md} cylindrical-trap-fig
<img src="Figure_2.jpg" alt="cylindrical trap"  width="600px">

Example of a cylindrical "trap" for measuring the energy density $u_\lambda$. Image credit: Carroll & Ostlie (2007).
```

For blackbody radiation,

```{math}
:label: energy_density
u_\lambda d\lambda = \frac{4\pi}{c}B_\lambda d\lambda = \frac{8\pi hc \lambda^{-5}}{e^{hc/(\lambda kT)}-1} d\lambda.
```

The total energy density $u$ is found by [integrating over *all* wavelengths](http://people.tamu.edu/~kevinkrisciunas/planck.pdf),

$$ u = \int_0^\infty u_\lambda d\lambda $$

and for blackbody radiation (using Eqn. {eq}`energy_density`), then total energy density is

```{math}
:label: tot_energy_density   
u = \frac{4\pi}{c} \int_0^\infty \frac{8\pi hc \lambda^{-5}}{e^{hc/(\lambda kT)}-1} d\lambda = \frac{4\sigma T^4}{c},
```

where $4\sigma/c = 7.565767 \times 10^{-16}\;{\rm J/m^3/K^4}$ and is known as the *radiation constant*.

### The Specific Radiative Flux
Astronomers are often interested in the *net* energy that passes through a *unit area* in the direction of the z-axis.  This is given by the **specific radiative flux** $F_\lambda$ as

```{math}
:label: spec_flux
F_\lambda d\lambda = \int I_\lambda d\lambda\:\cos\theta\:d\Omega = \int_{\phi = 0}^{2\pi} \int_{\theta = 0}^\pi I_\lambda d\lambda\:\cos\theta\sin\theta d\theta\:d\phi.
```

The specific radiative flux considers only the *z*-component (i.e., $\cos\theta$) of a light ray, where oppositely directed rays cancel.  Both the radiative flux and the specific intensity (Eqn. {eq}`specific_intensity`) describe the light received from a celestial source.  When a source is resolved by the telescope as defined by Rayleigh's criterion ($\theta > \theta_{\rm min}$), the *specific intensity* is measured.  For example, the measured value of the specific intensity (at 501 nm) at the center of the Sun's disk is $I_{501} = 4.03 \times 10^{13}\;{\rm W/m^{3}/sr}$.

```{figure-md} source-fig
<img src="Figure_3.jpg" alt="source"  width="600px">

The measurement of (a) the specific intensity for a resolved source and (b) the radiative flux for an unresolved source. Image credit: Carroll & Ostlie (2007).
```

For an unresolved source, the *radiative flux* is measured because the energy received from the entire source will disperse through the diffraction pattern (i.e., Airy disk and rings).  Because the light arriving at the detector leaves the source at all angles, the detector is effectively integrating the specific intensity over all directions.  As the distance to the source increases, the amount of energy falling within the Airy disk decreases as $1/r^2$, as expected.

### Radiation Pressure
Einstein's special theory of relativity describes photons as massless particles that nonetheless carry a momentum $p$ that is related to its energy $E$.  Thus, the photon can exert a **radiation pressure**.  Consider the gas pressure exerted by molecules bouncing off a wall, where the radiation pressure of photons can be described through a similar process. At a given time, a photon can strike a perfectly reflecting surface of area $dA$ and be reflected into a solid angle ${\rm d}\Omega$  (at the same angles $\theta$ and ${\rm d}\Omega$ due to the law of reflection).  The change in $p_z$ of photons within a range of wavelengths from the reflection is

$$ dp_{z,\lambda} d\lambda = p_\lambda \bigg\rvert_{z_o}^{z_f} d\lambda = \frac{2E_\lambda\cos\theta}{c} d\lambda = \frac{2}{c} I_\lambda d\lambda\:dt\:dA\cos^2\theta\:d\Omega, $$

using $E_\lambda$ from Eqn. {eq}`specific_intensity`.  Recall that $F = dp/dt$ and $P = F/A$ so that the radiation pressure $P_{\rm rad,\lambda}d\lambda$ can be written as

$$ P_{\rm rad,\lambda} d\lambda = \frac{2}{c} \int_{\phi = 0}^{2\pi} \int_{\theta = 0}^{\pi/2} I_\lambda d\lambda\:\cos^2\theta\sin\theta d\theta\:d\phi. \quad {\rm (reflection)} $$

```{figure-md} rad-press-fig
<img src="Figure_4.jpg" alt="radiation pressure"  width="500px">

Radiation pressure produced by reflected photons from the solid angle $d\Omega$. Image credit: Carroll & Ostlie (2007).
```

Instead of a reflecting wall, replace it with a mathematical surface $dA$ where the incident photons will now keep on going through and be streaming from the other side.  For an *isotropic radiation field*, the leading factor of 2 is removed because there is no longer a change in momentum upon reflection.  The radiation pressure becomes

```{math}
:label: rad_press_trans
P_{\rm rad,\lambda} d\lambda = \frac{1}{c} \int_{\phi = 0}^{2\pi} \int_{\theta = 0}^\pi I_\lambda d\lambda\:\cos^2\theta\sin\theta d\theta\:d\phi = \frac{4\pi}{3c}I_\lambda d\lambda, \quad {\rm (transmission)}
```

where the volume over all solid angles is simply a unit sphere ($4\pi/3$).  However, for a non-isotropic radiation field the equation for the radiation pressure is still valid, but the pressure depends on the orientation of the mathematical surface $dA$.

In general, the total radiation pressure at all wavelengths is found by

$$ P_{\rm rad} = \int_0^\infty P_{\rm rad,\lambda} d\lambda, $$

where the substitution $B_\lambda = I_\lambda$ in Eqn. {eq}`rad_press_trans` can be made blackbody radiation to get

\begin{align} 
P_{\rm rad} = \int_0^\infty B_\lambda(T)d\lambda = \frac{4\sigma T^4}{3c} = \frac{1}{3}u.
\end{align}

The **blackbody radiation pressure** is one-third of the energy density, where an ideal monatomic gas is two-thirds of its energy density.

### **Problems**
>1\. Find an expression for $n_\lambda\:d\lambda$ (the number density of blackbody photons) with a wavelength between $\lambda$ and $\lambda + d\lambda$.
>
>2\. Using your expression from Problem 1, find the total number of photons inside a kitchen oven assuming a volume of 0.5 ${\rm m^3}$ and set at 477 K.


## Stellar Opacity
The classification of stellar spectra is complicated by the fact that stars are not actually blackbodies.  The temperature of a star's photosphere can be estimated by the Stefan-Boltzmann law and the spectrum shape using the Planck function.  However, the Sun's spectrum deviates substantially from the idealized shape because solar absorption lines remove light at certain wavelengths.  A dense series of metallic absorption lines in the Solar spectrum is especially effective and called **line blanketing**.

### Temperature and Local Thermodynamic Equilibrium
There are many different measures of temperature within a star, which are defined according to a particular physical process.

- The temperature obtained from the Stefan-Boltzmann law (**effective temperature**) is *uniquely defined for a particular layer* of the stellar atmosphere.
- The **excitation temperature** is defined by the Boltzmann equation, which measures the ratio of atoms between two quantum states.
- The **ionization temperature** is defined by the Saha equation, which measures the ratio of atoms that are ionized (missing electrons).
- The **kinetic temperature** is contained in the Maxwell-Boltzmann distribution, which describes the velocity distribution of gas particles.
- The **color temperature** is obtained by fitting the Planck function to the shape of a star's continuous spectrum.

```{figure-md} solar-spec-fig
<img src="Figure_5.jpg" alt="solar spectra"  width="700px">

The spectrum of the Sun in 2 nm wavelength intervals. The dashed curve represents an ideal blackbody with the Sun's effective temperature. Image credit: Carroll & Ostlie (2007); data taken from [Aller (1991)](https://ui.adsabs.harvard.edu/abs/1991asn..book.....A/abstract).
```

The effective temperature is the only measure that depends strictly on the location within the star, while the remaining temperatures vary according to the conditions of the gas that can be described by the simple case of a gas confined within an "ideal" box (i.e., ideal gas law).  Due to the boundary condition of the box, the energy of the gas particles and blackbody radiation will come into equilibrium and can be described by a single well-defined temperature.  As a result, there is no net flow of energy through the box, or between the matter and radiation.  The number of photons absorbed per unit time is equal to the number of photons emitted, which is called [thermodynamic equilibrium](https://en.wikipedia.org/wiki/Thermodynamic_equilibrium).

Such a case cannot be true everywhere in a star (i.e., stars shine so some photons escape) and a star cannot be in perfect thermodynamic equilibrium.  A net outward flow of energy occurs through the star and the temperature varies with the stellar radius.  The ideal box approximation can be used: *if the distance over which the temperature changes significantly is large compared with the distances traveled by the particles and photons between collisions* (i.e., the ideal box is small enough).  The distances traveled by particles are called their *mean free paths*.  When the above condition is met, the particles and photons cannot escape the local environment and the temperature of the box is nearly constant (i.e., **local thermodynamic equilibrium** *LTE*).

The photosphere is the surface layer of the Sun's atmosphere where the photons can escape into space.  The temperature in one region varies from 5580 K to 5790 K over a distance of 25.0 km.  The *temperature scale height* $H_T$ is the characteristic distance over which the temperature varies: $H_T \equiv T_{\rm avg}/|dT/dr|$.

- **What is the temperature scale height $H_T$ of the Sun's photosphere?**

In [1]:
import numpy as np
def scale_height(T1,T2,r):
    #T1 = low temperature
    #T2 = high temperature
    #r = distance traveled
    T_avg = np.mean([T1,T2]) #average temperature
    dT = np.abs(np.diff([T1,T2])) #temperature range dT
    return T_avg/(dT/r)

T1_Sun = 5580 #K
T2_Sun = 5790 #K 
r_Sun = 25.0 #km
H_T = scale_height(T1_Sun,T2_Sun,r_Sun)
print("The scale height of the Sun's photosphere is %3d km." % np.round(H_T,0))

The scale height of the Sun's photosphere is 677 km.


**Is the Sun's photosphere in LTE?** To answer this question, we need to find the mean free path of the particles and photons.  The density of the Sun's photosphere is about $\rho = 2.1 \times 10^{-4}\:{\rm kg/m^3}$ and consists primarily of neutral hydrogen atoms in the ground state (see the [previous chapter](https://saturnaxis.github.io/ModernAstro/Chapter_8/classification-of-stellar-spectra.html#application-to-the-sun)).  The number density of hydrogen atoms (i.e., number of H atoms per cubic meter) is roughly

$$ n = \frac{\rho}{m_H} = 1.27 \times 10^{23}\:{\rm m^{-3}} $$

with the mass of hydrogen $m_H = 1.67 \times 10^{-27}\:{\rm kg}$. The collision probability for two atoms is higher if their centers pass within twice the atomic radius (i.e., [Bohr radius](https://saturnaxis.github.io/ModernAstro/Chapter_5/interaction-of-light-and-matter.html#bohr-s-semiclassical-atom)) or $2a_o$, where $a_o \approx 5.29 \times 10^{-11}\:{\rm m}$. This region of higher collision probability defines an area or **cross section** $\sigma \equiv \pi(2a_o)^2$.  


```{figure-md} mean-free-path-fig
<img src="Figure_6.jpg" alt="mean free path"  width="600px">

Mean free path $\ell$ of a hydrogen atom. Image credit: Carroll & Ostlie (2007).
```

Consider a single atom of radius $2a_o$ with a cross section of $4\pi a_o^2$ moving with a speed $v$ over a time $t$.  The atom has moved a distance $vt$ and sweeps out a cylindrical volume $V=\sigma vt$.  Within the volume, there are $n\sigma vt$ point atoms that collide with the moving atom.  The **mean free path** $\ell$ between collisions is

```{math}
:label: mean_free_path
\ell = \frac{\rm distance\:traveled}{\rm number\:of\:atoms} = \frac{vt}{n\sigma vt} = \frac{1}{n\sigma}.
```

- **What is the mean free path for the Sun's photosphere consisting of hydrogen atoms?**

In [ ]:
def mean_free_path(n,sigma):
    #n = number density of atoms
    #sigma = cross section
    return 1./(n*sigma)

n_H = 2.1e-4/1.67e-27 #number density of H atoms in Sun's photosphere
a_o = 5.29e-11 #m; radius of the Bohr atom
sigma_H = np.pi*(2*a_o)**2
ell_H =  mean_free_path(n_H,sigma_H)
print("The mean free path between collisions in the Sun's photosphere is %1.2e m." % np.round(ell_H,6))

The temperature scale height in the Sun's photosphere is 677 km and the mean free path for collisions is $2.26 \times 10^{-4}$ m, where $H_T/\ell \approx 3 \times 10^9$.  The criteria for LTE is satisfied for the hydrogen atoms of the photosphere (not the photons) and the gas particles see an essentially constant kinetic temperature between collisions.  In addition, they are effectively confined within a limited volume.  

### The Definition of Opacity
Any process that removes photons from a beam of light is collectively termed **absorption**, which includes the *scattering* of photons (i.e., Compton scattering), the absorption of photons by atomic electrons making upward transitions, and molecular energy-level transitions.  The change in intensity $dI_\lambda$ of a photon as it travels through a gas is proportional to its intensity $I_\lambda$, the distance traveled $ds$, and the density of the gas $\rho$ through the relation,

```{math}
:label: intensity_change
\frac{dI_\lambda}{I_\lambda} = -\kappa_\lambda \rho  ds.
```

The quantity $\kappa_\lambda$ is called the wavelength-dependent **absorption coefficient** (or **opacity**).  The intensity decreases (due to the minus sign) along the path $s$ traveled by the photon.  The opacity has units of area/mass (${\rm m^2/kg}$) because it is cross section of stellar gas that is removing the photons from the beam of light.  In general, the opacity of a gas is a function of its composition, density, and temperature.

Consider a monochromatic beam of light with initial intensity $I_o$ and wavelength $\lambda$ that travels a distance $s$.  The final intensity $I_f$ can be found by integrating both sides of Eqn. {eq}`intensity_change` by

$$ \int_{I_o}^{I_\lambda} \frac{dI}{I} = -\kappa \rho \int_0^s ds,$$

which has the solution,

$$ \ln{\frac{I_f}{I_o}} = -\kappa \rho s, $$

and is simplified by exponentiating both sides (and dropping the subscript $f$) to get

```{math}
:label: intensity_absorb
I = I_o e^{-\kappa \rho s},
```

which shows that the intensity exponentially decays as the light travels a distance $s$ into the medium described by the absorption coefficient $\kappa$ and density $\rho$.  There is a characteristic length $\ell = (\kappa \rho)^{-1}$ by which the intensity declines.  

- **What is the characteristic length $\ell$ for a photon near the peak wavelength (500 nm) in the Solar photosphere?** (Use $\kappa_{500} = 0.03\;{\rm m^2/kg}$. )

In [ ]:
def opacity_length(kappa,rho):
    #kappa = absorption coefficient at a given wavelength; m^2/kg
    #rho = density of the gas; kg/m^3
    return 1./(kappa*rho)

kappa_500 = 0.03
rho_photo_Sun = 2.1e-4
ell_500 = opacity_length(kappa_500,rho_photo_Sun)/1000 #convert to km
print("The characteristic distance traveled by a photon in the Solar photosphere is %3d km." % np.round(ell_500,-1))

The temperature scale height is 677 km, where for photons $H_T/\ell \approx 4.2$ and the criteria for local thermodynamic equilibrium is not satisfied.  The photospheric photons do *not* see a constant temperature and LTE is not strictly valid in the photosphere.  Although LTE is a commonly invoked assumption in stellar atmospheres, it *must* be used with caution.  

### Optical Depth
Scatter photons have a mean free path (given by Eqn. {eq}`mean_free_path`), where $\kappa_\lambda \rho$ and $n\sigma_\lambda$ can be though of as the *fraction* of photons scatter per meter of distance traveled.  Note that the mean free path is dependent on the photon wavelength $\lambda$.  However it is convenient to define an **optical depth** $\tau_\lambda$ as 

\begin{align} d\tau_\lambda = -\kappa_\lambda \rho ds, \end{align} 

where $s$ is the distance measured along the photon's direction of motion (i.e., looking back along the path traveled by the photon).  The difference in optical depth $\Delta \tau_\lambda$ can be determined as the difference between the final ($\tau_{\lambda,f}$) and initial ($\tau_{\lambda,o}$) optical depths, or the integral along the photon's path as

```{math}
:label: tau_diff
\Delta \tau_\lambda = \tau_{\lambda,f} - \tau_{\lambda,o} = -\int_0^s \kappa_\lambda \rho ds.
```

As the light approaches the observer, it travels from a deeper layer (greater $\tau_\lambda$) to shallow layer (lesser $\tau_\lambda$) and $\Delta \tau_\lambda < 0$.  The outermost layers of a star have $\tau_\lambda = 0$ for all wavelengths since the light is leaving the star's atmosphere.  In this case, Eqn. {eq}`tau_diff` becomes

```{math}
:label: tau_integral
\tau_\lambda = \int_0^s \kappa_\lambda \rho ds
```

and the subscript $o$ is dropped.

```{figure-md} optical-depth-fig
<img src="Figure_7.jpg" alt="optical depth"  width="450px">

Optical depth $\tau_\lambda$ measured along a ray's path starting from outside the material. Image credit: Carroll & Ostlie (2007).
```

The optical depth can be related to the intensity by combining Eqn. {eq}`intensity_absorb` with Eqn. {eq}`tau_integral`, where there is a decline in the intensity given by

$$ I_\lambda = I_{\lambda,o} e^{-\tau_\lambda}.$$

If the initial optical depth is equal to unity ($\tau_{\lambda,o} = 1$), then the intensity will decline by a factor of $1/e$ before escaping into the vacuum of space.

```{note}
The optical depth can be described as the number of mean free paths the photon travels starting from the original position to the surface.
```
The maximum optical depth is typically $\tau_\lambda \approx 1$, where deeper layers are completely obscured.  In some situations, setting $\tau_\lambda = 0$ occurs inside the star instead of at the surface.  

- If $\tau_\lambda \gg 1$, then the gas is *optically thick*.
- If $\tau_\lambda \ll 1$, then the gas is *optically thin*. 

The optical depth varies with wavelength, where a gas could be optically thick for a specific wavelength but optically thin for a different wavelength.  For example, the Earth's atmosphere is optically thick for X-rays, but optically thin for visible light.

```{figure-md} ray-absorption-fig
<img src="Figure_8.jpg" alt="ray absorption"  width="600px">

(a) A light ray entering Earth's atmosphere at an angle $\theta$ (b) Example of the determination of $I_{\lambda,o}$ from a fit of $I_\lambda$ vs. $\sec \theta$. Image credit: Carroll & Ostlie (2007).
```

- **How can the opacity of the Earth's atmosphere be corrected for?**

The intensity of the light before entering the atmosphere is $I_{\lambda,o}$ and $I_\lambda$ once it reaches the telescope.  The optical depth is measured backwards along the path traveled by the light and thus, $\tau_\lambda = 0$ at the telescope.  The distance traveled is the height $h$ of the atmosphere.  The light enters the atmosphere at some angle $\theta$, where $ds = -dz/\cos \theta = -\sec \theta dz$.  Using Eqn. {eq}`tau_integral` we get

$$ \tau_\lambda = \int_0^s \kappa_\lambda \rho ds = \sec \theta \int_0^h \kappa_\lambda \rho dz = \tau_{\lambda,o} \sec \theta, $$

where $\tau_{\lambda,o}$ refers to the optical depth of a vertically traveling photon ($\theta = 0).  The intensity of the light at the telescope is 

\begin{align} 
I_\lambda = I_{\lambda,o}e^{-\tau_{\lambda,o}\sec \theta}.
\end{align}

For a single observation, both $I_{\lambda,o}$ and $\tau_{\lambda,o}$ are unknown.  Over a night of observation, the Earth rotates on its axis and several measurements of the received intensity $I_\lambda$ as a function of $\sec \theta$ can be made.  Then the best-fitting *slope* is $-\tau_{\lambda,o}$ and $I_{\lambda,o}$ can be determined through extrapolation 

$$ \lim_{\sec \theta \rightarrow 0} \ln I_\lambda = \ln I_{\lambda,o}. $$  

Measurements of the specific intensity or radiative flux can be corrected for absorption by Earth's atmosphere.

### General Sources of Opacity
The opacity of a material depends on the chemical composition and how photons interact with the particles (atoms, ions, and free electrons) within the material.  If a photon passes within $\sigma_\lambda$ (i.e., the particles's cross-sectional area) of the particle, then the phone can be either: scattered or absorbed.

- In an absorption process, the photon ceases to exist and its energy is incorporated into the thermal energy of the gas.

- In a scattering process, the photon just continues along in a different direction.

Both absorption and scattering can remove photons (of a specific wavelength) like a sieve and contribute to the opacity $\kappa_\lambda$ of the stellar material.  If the opacity varies slowly with wavelength, it contributes to the star's continuous spectrum (or *continuum*) and dark absorption lines are the result of a *rapid* variation in the opacity.

There are four primary sources of opacity available for removing photons from our observations of a star.  These sources involve the quantum state of an electron, where the terms *bound* and *free* are used to describe the initial and final states of the electron within an atom.

1. **Bound-bound transitions** are electron transitions from one orbital to another (i.e., excitation and de-excitation).  The bound-bound opacity $\kappa_{\lambda,bb}$    accounts for electrons that remove photons through absorption and the re-emission of the photons are scattered into a different direction than the line-of-sight.  This opacity is small except at wavelengths capable of absorption and hence is responsible for absorption lines in stellar spectra.  

    The de-excitation may also come from collisions with neighboring particles.  An important by-product is the degrading of the average energy of the photons.  For example, if two photons are emitted (from a cascade process) for every photon absorbed, then the average energy is cut in half.  There is not simple equation for bound-bound transitions.

2. **Bound-free absorption**, or *photoionization*, occurs when an incoming photon has enough energy to free, or ionize, an atom.  The free electron can have any energy when removed from the atom, but the incoming photon must have a wavelength $\lambda \leq hc/\chi_n$, where $\chi_n$ is the ionization energy of the $n{\rm th}$ orbital.  The bound-free opacity $\kappa_{\lambda,bf}$ contributes to the continuum opacity.

    The photoionization cross section for the hydrogen atom is  <br><br>

    $$ \sigma_{bf} = \frac{1.31 \times 10^{-19}}{n^2} \left(\frac{\lambda}{500\:{\rm nm}}\right)^3 \:{\rm m^2}, $$

    which depends on the quantum state $n$ and the wavelength $\lambda$.  The inverse process of *free-bound* emission occurs during atomic recombination (free electron recombines with an ion, emitting a photon).  Similar to bound-bound emission, this also reduces the average energy of the photons.

3. **Free-free absorption** is a scattering process that occurs when the speed of a free electron increases (or decreases) after a nearby ion absorbs (or emits) a photon, respectively.  This process occurs due to conservation of both energy and momentum.  The free-free opacity $\kappa_{\lambda,ff}$ contributes to the continuum opacity.  The process of free-free emission is also known as [bremsstrahlung](https://en.wikipedia.org/wiki/Bremsstrahlung), when means "braking radiation" in German.

    ```{figure-md} ff-absorption-fig
    <img src="Figure_9.jpg" alt="ray absorption"  width="450px">

    Free-free absorption when an free electron is scattered in the vicinity of an ion due to a photon. Image credit: Carroll & Ostlie (2007).
    ```
4. **Electron scattering** is as you would expect where a photon is scatter (*not absorbed*) by a free electron through the process of [Thomas scattering](https://en.wikipedia.org/wiki/Thomson_scattering).  The electron "oscillates" (like a buoy) in the wake of a photon.  Because the electron's cross section is tiny, it is a poor target for an incident photon.  The cross section for Thomson scattering $\sigma_T$ has the same value (independent of wavelength),  <br> <br>

    \begin{align} \sigma_T = \frac{1}{6\pi\epsilon_o^2}\left( \frac{e^2}{m_e c^2}\right) = 6.65 \times 10^{-29}\:{\rm m^2}. \end{align} 

    This is ~2 billion times smaller than the hydrogen cross section for photoionization from bound-free absorption, $\sigma_{bf}$.  Electron scattering is most effective when the electron density is very high (i.e., high temperatures for ionization) and other sources of opacity are eliminated due to the absence of necessary bound electrons.  For high temperatures, the electron scattering opacity $\kappa_{es}$ dominates the continuum opacity. <br> <br>

    A photon can also be scattered by loosely bound electrons.  *Compton scattering* occurs when the photon's wavelength is much smaller than the atom or *Rayleigh scattering* if the wavelength is much larger.  Compton scatter is usually lumped together with Thomson scattering due to the small change in wavelength of the scattered photon.  The cross section for Rayleigh scattering is smaller than the Thomson cross section and can be neglected in most stellar atmospheres.  However, it is important in the UV for supergiant stars with their extended envelopes and in cool main-sequence stars.
    The opacity of stellar material suddenly increases at wavelengths $\lambda \leq 364.7\:{\rm nm}$ and the measured radiative flux of the star suddenly decreases.  This abrupt drop in the continuous spectrum is the **Balmer jump** and is evident in the Sun's spectrum.  The size of the Balmer jump in hot stars depends on the fraction of hydrogen atoms in the first excited state.  The Balmer jump appears in the middle of the *U* band in the *UBV* system, which results in a decrease in the amount of light received in the bandwidth of the *U* filter and increase the measured values for both the *U* magnitude and color index ($U-B$).

5. **$H^-$ ion photoionization** occurs in the atmospheres of stars later than F0 and is the primary source of continuum opacity, $\kappa_{H^-}$.  An $H^-$ ion is a hydrogen atom with a second electron, where there is partial shielding provided by the nucleus (to prevent repulsion) and thus the second electron can be loosely bound on the opposite side of the atom.  The binding energy (0.754 eV) is small compared with the binding energy of the hydrogen atom from the ground state.  The electron is easily liberated by photons with energy greater than 0.754 eV, where the excess energy is converted to kinetic energy.  On the other hand, a hydrogen atom can also capture a second electron and release a photon (proportional to the excess kinetic energy of the captured electron) in the process (i.e., $H+e^- \rightleftharpoons H^- + \gamma$).  A photon with an energy equal to 0.754 eV corresponds to a wavelength of 1640 nm.  At longer wavelengths $H^-$ contributes to opacity through free-free absorption (i.e., more free electrons become available through photoionization).

```{note}
Rayleigh scattering is important in planetary atmospheres and is responsible for Earth's blue sky.
```
- **What is the maximum wavelength photon that can ionize a hydrogen atom in the first excited state?**

In [ ]:
def bound_free_H(n):
    #n = the energy state of hydrogen (1 = ground, 2 = first excited, etc.)
    chi_n = 13.6/n**2 #eV; ionization energy
    return hc_eV/chi_n #return max wavelength in nm

hc_eV = 1239.84198 #eV*nm; https://en.wikipedia.org/wiki/Planck_constant
print("The maximum wavelength photon that can ionize a H atom (first excited state) is %3.1f nm." % np.round(bound_free_H(2),1))

### The Rosseland Mean Opacity
The opacity in Eqn. {eq}`tau_integral` depends on the wavelength of light, where a star is expected to produce photons covering a wide range of wavelengths.  It is often useful to define an opacity that has been averaged over all wavelengths.  This averaging produces an opacity that depends only on the composition, density, and temperature of the stellar atmosphere.  The most commonly used scheme to compute a wavelength-independent opacity is the [Rosseland mean opacity](http://personal.psu.edu/rbc3/A534/lec6.pdf) or the **Rosseland mean**.  The Rosseland mean incorporates a weighting function that depends on how fast the blackbody spectrum varies with temperature, which is defined as

\begin{align} \frac{1}{\bar{\kappa}} \equiv \frac{\int_0^\infty \frac{1}{\kappa_\lambda}\frac{\partial B_\lambda(T)}{\partial T} d\lambda}{\int_0^\infty \frac{\partial B_\lambda(T)}{\partial T} d\lambda}. 
\end{align}

The absorption coefficient $\kappa_\lambda$ varies with wavelength, but it also depends on the assumed source of opacity (e.g., bound-free, free-free, etc.) and there is not a simple equation to describe the average absorption coefficient for bound-bound transitions.  Approximation formulae have been developed for the average bound-free and free-free opacities:

```{math}
:label: bf_kappa
\bar{\kappa}_{bf} = 4.34 \times 10^{21} \frac{g_{bf}}{t}Z(1+X)\frac{\rho}{T^{3.5}} \: {\rm m^2/kg}
```
```{math}
:label: ff_kappa
\bar{\kappa}_{ff} = 3.68 \times 10^{18}\:g_{ff}(1-Z)(1+X)\frac{\rho}{T^{3.5}} \: {\rm m^2/kg},
```

which depends on the density $\rho$, temperature $T$, the hydrogen **mass fraction** $X$, helium mass fraction $Y$, and the fractional abundance (by mass) of metals $Z$.

```{note}
Because most stellar gases are primarily composed of hydrogen and helium, all other constituents are lumped together and referred to as *metals*.  However, certain applications require more specific detail in the composition, where each species is represented by its own mass fraction.
```

The mass fractions are formally defined as

\begin{align}  X \equiv \frac{\rm total\: mass\:of\:hydrogen}{\rm total\: mass\:of\:gas} \end{align}
\begin{align}  Y \equiv \frac{\rm total\: mass\:of\:helium}{\rm total\: mass\:of\:gas} \end{align}
\begin{align} Z \equiv \frac{\rm total\: mass\:of\:metals}{\rm total\: mass\:of\:gas} \end{align}

such that $X+Y+Z = 1$.  The [Gaunt factors](https://en.wikipedia.org/wiki/Gaunt_factor) ($g_{bf}$ and $g_{ff}$) in Eqns. {eq}`bf_kappa` and {eq}`ff_kappa` are quantum-mechanical correction terms.  These Gaunt factors are approximately 1 for the visible and UV wavelengths of interest.  The **guillotine factor** $t$ ([Eddington 1932](https://ui.adsabs.harvard.edu/abs/1932MNRAS..92..364E/abstract)) describes a cutoff to the opacity of an atom after it has been ionized, where typical values lie between 1 and 100.

Equations {eq}`bf_kappa` and {eq}`ff_kappa` have the functional form $\bar{\kappa} \propto \kappa_o \rho/T^{3.5}$, where $\kappa_o$ combines all the factors that depend on composition. H. A. Kramers derived the first forms (in 1923) using classical physics and the Rosseland mean.  As a result, any opacity having a temperature and density dependence is referred to as **Kramers opacity law**.

The electron scattering cross section is independent of wavelength and the Rosseland mean is simply

\begin{align} \bar{\kappa}_{es} = 0.02(1+X) \:{\rm m^2/kg}. \end{align}

The mean opacity is modified in cooler stars (later than F0) by the $H^-$ ion over specific temperature range (3000 K $\leq T \leq$ 6000 K), density range ($10^{-7}\leq \rho \leq 10^{-2}\: {\rm kg/m^3}$), and metal mass fraction ($0.001 < Z < 0.03$), when the hydrogen mass fraction $X\sim0.7$ is lower.  The average opacity is

\begin{align} \bar{\kappa}_{H^-} \approx 7.9 \times 10^{-34} (50Z)\sqrt{\rho}T^9\:{\rm m^2/kg}. \end{align}

The total Rosseland mean opacity $\bar{\kappa}$ is the average of the sum of the individual contributions or

$$ \bar{\kappa} = \overline{\kappa_{bb}+\kappa_{bf}+\kappa_{ff}+\kappa_{es}+\kappa_{H^-}}. $$

```{figure-md} Rosseland-mean-fig
<img src="Figure_10.jpg" alt="Rosseland mean opacity"  width="500px">

Rosseland mean opacity for a star with $X=0.7$, $Y=0.28$, and $Z=0.02$. The curves are labeled by the logarithmic value of the density. Image credit: Carroll & Ostlie (2007); data taken from [Iglesias & Rogers (1996)](https://ui.adsabs.harvard.edu/abs/1996ApJ...464..943I/abstract).
```

From detailed computer calculations of the Rosseland mean using quantum physics, we find that the opacity increases linearly with density for a given temperature.  At a contour of constant density, the opacity rises with increasing temperature that reflects the greater number of free electrons produce by ionization of hydrogen and helium.  The hydrogen partial ionization zone and neutral helium ionization occur at about the same temperature (10,000 K).  The decline in opacity (after $\log_{10} T = 4$ or 10,000 K) roughly follows a Kramers law ($\bar{\kappa} \propto T^{-3.5}$) and is primarily due to the bound-free and free-free absorption.

The He II ion becomes fully ionized at 40,000 K for a wide range of stellar parameters, where there is a slight increase in the number of free electrons available.  Ionization of certain metals (most notably iron) occurs above $10^5$ K, which produces another increase in the average opacity.  Electron scattering dominates at the highest temperatures when nearly all the stellar material is ionized (i.e., few electrons available for bound-bound or bound-free processes).

### **Problems**
>3\. Take the radius of a nitrogen molecule ${\rm N}_2$ to be 0.1 nm and the density of air to be 1.2 ${\rm kg/m^3}$.  A nitrogen molecule contains 28 nucleons (protons and neutrons).
>>**(a)** Use the root-mean-square speed, $v_{\rm rms}$, estimate the mean free path of the nitrogen molecules in your classroom at room temperature (300 K).  
>>**(b)** What is the average time between collisions?
>
>4\. Calculate how far you could see through Earth's atmosphere if it had the opacity of the solar photosphere ($\kappa_{500}=0.03\:{\rm m^2/kg}$).
>
>5\. According to one model of the Sun, the central density is $1.53 \times 10^5 \:{\rm kg/m^3}$ and the Rosseland mean opacity at the center is 0.217 ${\rm m^2/kg}$.
>>**(a)** Calculate the mean free path of a photon at the center of the Sun.  
>>**(b)** Calculate the average time it would take for the photon to escape from the Sun if this mean free path remained constant for the photon's journey to the surface.

## Radiative Transfer
This section describes stellar atmospheres that are in equilibrium or a steady-state.  This means that the mechanisms involved in absorbing and emitting energy must be in balance throughout the star so that no change in total energy occurs within any layer of the interior.  There are stars where this is not the case (e.g., pulsating stars) and such stars periodically absorb or "dam up" the outward flow of energy to drive stellar oscillations.

### Photon Emission Processes
**Emission** is defined as any process that adds photons to a beam of light, which includes the scattering of photons into the beam and the true emission of photons through electron transitions.  Opacity described processes (bound-bound and free-bound) that absorb photons, where there is an inverse process through emission (e.g., bound-bound emission, etc.).  The processes of absorption and emission hinder the flow of photons through a star and there is not a direct flow of photons streaming towards the surface.  Instead, individual photons travel only temporarily within the beam as they are repeatedly scattered in random directions (i.e., going on side-quests) while on their journey.

### The Random Walk
As the photons move through the stellar material, they follow a **random walk** (i.e., a haphazard path).  Between each atomic encounter, the photon travels a path ${\bf \ell}$ (the mean free path), which can be summed over a large number $N$ directed steps to produce a net vector displacement **d**:

$$ {\bf d} = {\bf \ell}_1 + {\bf \ell}_2 + \cdots + {\bf \ell}_N = \sum_{i=1}^N {\bf \ell}_i. $$

```{figure-md} displacement-fig
<img src="Figure_11.jpg" alt="displacement"  width="450px">

Displacement **d** of a random walking photon. Image credit: Carroll & Ostlie (2007).
```

The vector dot product of **d** determines the scalar contribution of each path by

$$ \begin{align*}
{\bf d}\cdot{\bf d} &={\bf \ell}_1 \cdot {\bf \ell}_1 + {\bf \ell}_1 \cdot {\bf \ell}_2 + \cdots {\bf \ell}_1 \cdot {\bf \ell}_N \\
&\quad+ {\bf \ell}_2 \cdot {\bf \ell}_1 +{\bf \ell}_2 \cdot {\bf \ell}_2 + \cdots + {\bf \ell}_2 \cdot {\bf \ell}_N \\
&\quad+ \cdots +{\bf \ell}_N \cdot {\bf \ell}_1 +{\bf \ell}_N \cdot {\bf \ell}_1 + \cdots + {\bf \ell}_N \cdot {\bf \ell}_N \\
&= \sum_{i=1}^N \sum_{j=1}^N {\bf \ell}_i \cdot {\bf \ell}_j.
\end{align*}$$

Using the magnitude definition of a vector dot product ($A\cdot B = |A||B|\cos\theta$) and each vector has the same length ($|{\bf \ell_i}|^2 = |{\bf \ell_j}|^2 = \ell^2$), then we have $N$ diagonal terms ($i=j$) and $N(N-1)$ off-diagonal ($i\neq j$) terms.  Along the diagonal, $\cos\theta_{ii} = \cos\theta_{jj} = 1$, so that we find

$$ d^2 = {\rm sum\:of\:diagonal\:terms} +  {\rm sum\:of\:off-diagonal\:terms} $$

or

$$ d^2 = N\ell^2 + \ell^2  \sum_{i=1}^N \sum_{j=1}^N \cos \theta_{ij},\quad {\rm for\:}i\neq j. $$

For a large number of randomly directed steps, the sum of all the cosine terms approaches zero.  For a random walk, the displacement $d$ scales with the size of each step $\ell$ by

```{math}
:label: step_size
d = \ell \sqrt{N}.
```

Through a random walk, the transport of energy through a star by radiation alone can be extremely inefficient.  It takes 100 steps for a photon to travel a distance of $10\ell$ using Eqn. {eq}`step_size`.  The optical depth is roughly the number of mean free paths $\ell$ that a photon must travel to get to the surface, which implies that $d = \tau_\lambda \ell = \ell\sqrt{N}$.  The average number of steps needed to travel to a distance $d$ is

```{math}
:label: number_steps
N = \tau_\lambda^2,
```

for $\tau_\lambda \gg 1$.  When $\tau_\lambda \approx 1$, a photon can potentially escape from that depth, but a more carful analysis shows the average depth of the atmosphere from which photons can escape is $\tau_\lambda = 2/3$.

```{note}
Looking into a star at any angle, we can look back to an optical depth of about $\tau_\lambda = 2/3$ as measured relative to the line of sight.
```

A star's photosphere is defined as the layer which visible light can escape the star's atmosphere and come to us or $\tau_\lambda \approx 2/3$ for wavelengths in the star's continuum.


### Limb Darkening
An observer looking at the Sun's disk sees the brightest (hottest) portion near the center due to optical depth ($\tau_\lambda = 2/3$), observations near the edge (or *limb*) of the Sun appear darker (cooler).  The optical depth limits how deep an observer can see into the Sun, where at the center a finite depth reaches the hotter inner layers that glow more brightly.  At the limb, the penetration depth only reaches the uppermost (cooler) layers of the atmosphere and we will see a lower temperature compared to the center of the disk.  This effect is called **limb darkening**, where astronomers have to account for it in transit photometry (e.g., binary stars and [exoplanets](https://www.paulanthonywilson.com/exoplanets/exoplanet-detection-techniques/the-exoplanet-transit-method/)).

```{figure-md} limb-darkening-fig
<img src="Figure_12.jpg" alt="limb darkening"  width="450px">

Schematic of limb darkening, where the brightness of the stellar disk depends on the temperature of the layer that is reached within the stellar atmosphere along the line of sight. The thickness of a typical photosphere is ~0.1% of the stellar radius. Image credit: Carroll & Ostlie (2007).
```


### The Radiation Pressure Gradient
In addition to the photon's meandering journey, there is a pressure that builds within the Sun's atmosphere due to collisions.  This is similar to the air pressure within a closed room, where an individual molecule moves ~500 m/s and collides with other air molecules several billion times per second.  In the analogy of the closed room, a net flow of air (i.e., breeze) can occur after opening a window.  

The pressure outside is lower than the pressure inside the room, which can be expressed as a pressure differential using Bernoulli's principle.  In a star, the high pressure interior and vacuum of space also produces a pressure difference.  The extent of the difference is not uniform, where the pressure decreases with stellar radius (i.e., as you get closer to the surface).  But it is enough to create a light movement of photons toward the surface that described by

\begin{align}  \frac{dP_{\rm rad}}{dr} = -\frac{\bar{\kappa}\rho}{c}F_{\rm rad}. \end{align} 

The transfer of radiation as a flow with photons drifting toward the surface is a better description than a "beam" that propagate radially outward.  The description of a beam is a convenient fiction used to define the direct of motion for a photon at a snapshot along its journey of continual absorption and scattering.

## The Transfer Equation
### The Emission Coefficient
When examining the transfer of radiation, we are interested in the net flow of energy, not the specific path taken by individual photons. The intensity of light (i.e., change in flow) can be increased through the emission process as a ray travels through a gas.  The change in intensity $dI_\lambda$ is proportional to the distance traveled $ds$ and the density of the gas $\rho$.  Pure emission is described by

```{math}
:label: emission_coeff
dI_\lambda = j_\lambda \rho ds, 
```

where $j_\lambda$ is the wavelength-dependent **emission coefficient** of the gas in units of ${\rm m/s^3/sr}$.  Unlike the definition for the absorption coefficient, the contribution due to emission does not depend on $I_\lambda$ and thus the emission coefficient $j_\lambda$ has different units than the absorption coefficient $\kappa_\lambda$.

The specific intensity $I_\lambda$ of a beam changes as photons are added (emission or scattering into) or removed (absorption or scattering out) from the beam. Combining the contributions from both absorption and emission gives the general result

```{math}
:label: beam_intensity
dI_\lambda = -\kappa_\lambda \rho I_\lambda ds + j_\lambda \rho ds = (-\kappa_\lambda I_\lambda + j_\lambda)\rho ds.
```

The competing processes (in Eqn. {eq}`beam_intensity`) determine the changes in the beam intensity.  Imagine an interstate highway, where there is a flow of cars in one direction.  Due to the placement of on- and off-ramps, the total number of cars can increase or decrease over time.  The instantaneous changes in the flow of cars onto/off-off the highway affect the net flow of cars, where a large influx can lead to a traffic jam.

### The Source Function and the Transfer Equation
In a stellar atmosphere or interior, photons are added or removed from a net flow of radiation and we observe the net result.  The ratio of emission and absorption rates determine how rapidly the beam intensity changes and the tendency for a specific wavelength of photon to dominate the local population.  To determine the ratio of emission to absorption ($j_\lambda/\kappa_\lambda$), or **source function** $S_\lambda$ , we divide Eqn. {eq}`beam_intensity` by $-\kappa_\lambda \rho ds$:

```{math}
:label: source_fcn
-\frac{1}{\kappa_\lambda \rho}\frac{dI_\lambda}{ds} = I_\lambda -  \frac{j_\lambda}{\kappa_\lambda} = I_\lambda - S_\lambda.
```

The source function has the same units of intensity (${\rm W/m^3/sr}$) and is one form of the **equation of radiative transfer** or **transfer equation**.  According to the transfer equation, the intensity of light will not vary if $dI_\lambda/ds = 0$ and $I_\lambda = S_\lambda$.

```{note}
This form of the transfer equation assumes that the atmosphere is in a steady state.  Otherwise, a time-derivative term would have to be included.  Recall the definition of $S_\lambda$ is inversely proportional to $\kappa_\lambda$ so that the source function "blows up" as $\kappa_\lambda \rightarrow 0$.  In this case, Equation {eq}`emission_coeff` should be used instead.
```

If the intensity is *greater* than the source function ($I_\lambda > S_\lambda$), then the intensity of light decreases with distance (i.e., more absorption than emission).  On the other hand, if the intensity is *less** than the source function ($I_\lambda < S_\lambda$), then the intensity increases with distance (i.e., more emission than absorption).  This is a mathematical restatement of: *the intensity of the light tends to become equal to the local value of the source function*.

### The Special Case of Blackbody Radiation
Consider a box containing an optically thick gas maintained at a constant temperature $T$. The confined particles and blackbody radiation are in thermodynamic equilibrium by definition because there isn't a flow of energy through the box or between the gas particles.  A consequence of this equilibrium is that every process of absorption is met with an inverse process of emission.  The intensity of the radiation is described by the Planck function $I_\lambda = B_\lambda$ and the variation of intensity goes to zero, $dI_\lambda/ds = 0$ and $S_\lambda = I_\lambda$.  Using the transitive property ($A=B$, $B=C$, and therefore $A=C$) and *thermodynamic equilibrium*, *the source function is equal to the Planck function*, $S_\lambda = B_\lambda$.

A star cannot be in perfect thermodynamic equilibrium because there is a net flow of energy from the center to the surface.  But deep the in atmosphere (or a large optical depth, $\tau_\lambda \gg 1$), a random-walking photon (with many scattering events) will take at least $\tau_\lambda^2$ steps (Eqn. {eq}`number_steps`) to reach the surface.  When the photon mean free path is small compared to temperature scale height, the photons are *effectively* confined to a volume of nearly constant temperature and the conditions for local thermodynamic equilibrium (LTE) are satisfied.  Overall the statements: **(a)** $I_\lambda = B_\lambda$ and **(b)** $S_\lambda = B_\lambda$ indicate that the radiation field is described by the Planck function and the source function describes the physical source of the radiation ($j_\lambda/\kappa_\lambda$) as one that produces the radiation.

**Applying the Transfer Equation**

Imagine a beam of light of initial intensity $I_{\lambda,o}$ entering a volume of gas of constant density $\rho$, constant opacity $\kappa_\lambda$, and a constant source function $S_\lambda$.  The intensity function $I_\lambda$ depends on the penetration depth $s$ of the volume, where Eqn. {eq}`source_fcn` can be written as a 1st order ordinary differential equation:

$$ \frac{dI_\lambda}{ds} = -\kappa_\lambda \rho (I_\lambda - S_\lambda). $$

Using the separation of variables results in:

$$ \frac{dI_\lambda}{I_\lambda - S_\lambda} = -\kappa_\lambda \rho ds, $$

and integrating both sides to get,

$$ \ln(I_\lambda - S_\lambda) - {\rm Const.} = -\kappa_\lambda \rho s. $$

The beam of light enters with an initial intensity $I_{\lambda,o}$ when $s=0$, so that the integration constant equals $e^{(I_{\lambda,o}-S_\lambda)}$.  Then the factor due to the integration constant can be combined into the logarithm as

$$ \ln \left(\frac{I_\lambda - S_\lambda}{I_{\lambda,o} - S_\lambda}\right) = -\kappa_\lambda \rho s, $$

which simplifies to

\begin{align}  I_\lambda(s) = I_{\lambda,o} e^{-\kappa_\lambda \rho s} + S_\lambda(1-e^{-\kappa_\lambda \rho s}). \end{align}

- **Suppose the source function $S_\lambda$ is twice the initial intensity $I_{\lambda,o}$, then plot the transformation of the intensity in units of $\kappa_\lambda \rho s$.**

In [ ]:
import numpy as np
import matplotlib.pyplot as plt 

def Intensity(x,S_fact):
    #x = k \rho s 
    #S_fact = S_\lambda/I_{\lambda,o}
    return np.exp(-x) + S_fact*(1.-np.exp(-x))

fs = 'x-large'
x_rng = np.arange(0,6,0.01)

fig = plt.figure(figsize=(5,5),dpi=150)
ax = fig.add_subplot(111)

ax.axhline(2,linestyle='--',color='r',lw=2)
ax.plot(x_rng,Intensity(x_rng,2),'k-',lw=2)

ax.set_xlim(0,6)
ax.set_ylim(0.8,2.1)

ax.set_xlabel("$\\kappa_\\lambda \\rho s$",fontsize=fs)
ax.set_ylabel("$I_\\lambda/I_{\\lambda,o}$",fontsize=fs);

### The Assumption of a Plane-Parallel Atmosphere
The intensity of light must depend on the path for the outward flow of energy.  The absorption and emission coefficients are the same for light traveling in all directions, but these coefficients depend on the temperature and density in a complicated way.  For astronomers to understand the stellar interior, they must know *at what depth* a spectral line is formed.  Since we are interested in the depth, let's rewrite the transfer equation in terms of the optical depth, which is simply

\begin{align} \frac{dI_\lambda}{d\tau_\lambda} = I_\lambda - S_\lambda. \end{align} 

Optical depth does not provide a unique geometric depth in the atmosphere and thus, must be replaced with a more meaningful measure of position.  The atmospheres of main-sequence stars are physically thin compared with the size of the star.  The atmosphere's radius of curvature is much larger than its thickness, where we may consider the atmosphere as a *plane-parallel slab*.  If we assume align the *z*-axis with the vertical direction, then $z=0$ denotes the top of this **plane-parallel** atmosphere and a **vertical optical depth** $\tau_{\lambda,v}(z)$ is defined as

\begin{align}  \tau_{\lambda,v}(z) \equiv \int_z^0 \kappa_\lambda \rho dz. \end{align} 

```{note}
The $v$ symbol is not to be confused with $\nu$, where the former signifies the **v** in vertical.
```

Recall the discussion for correcting the [opacity of the Earth's atmosphere](https://saturnaxis.github.io/ModernAstro/Chapter_9/stellar-atmospheres.html#optical-depth), the depth was parameterized in a similar manner.  A ray that travels upward at an angle $\theta$ has farther to go through the atmosphere to reach the surface.  The optical depth $\tau_\lambda$ is *greater* than the vertical optical depth $\tau_{\lambda,v}$.  The two optical depths are related by

\begin{align}  \tau_\lambda = \frac{\tau_{\lambda,v}}{\cos \theta} = \tau_{\lambda,v} \sec \theta \end{align} 

and then the transfer equation becomes

\begin{align}  \cos \theta \frac{dI_\lambda}{d\tau_{\lambda,v}} = I_\lambda - S_\lambda \end{align} 

for the approximation of a plane-parallel atmosphere.

```{figure-md} pp-atmos-fig
<img src="Figure_14.jpg" alt="plane parallel atmosphere"  width="450px">

Plane-parallel stellar atmosphere. Image credit: Carroll & Ostlie (2007).
```

The value of the vertical optical depth is wavelength dependent, but when considering the bulk flow of photons a *gray* atmosphere is assumed.  Gray atmospheres are used when the measured flux does not show a strong wavelength dependence.  Thus, the transfer equation for a plane-parallel gray atmosphere is 

\begin{align}  \cos \theta \dfrac{dI}{d\tau_v} = I - S. \end{align} 

Formulating the transfer equation in terms of an angle and optical depth leads to two useful relations describing the radiation field.

**1) Integrating over all solid angles**

The source function $S$ depends only on the local conditions, so we can integrate over all solid angles to get

\begin{align}  \frac{d}{d\tau_v}\int I \cos \theta d\Omega = \int I d\Omega - S \int d\Omega.  \end{align} 

This expression can be simplified using $\int d\Omega = 4\pi$, definition of $F_{rad}$ (Eqn. {eq}`spec_flux`, integrated over $\lambda$) and the mean intensity (Eqn. {eq}`mean_intensity`, integrated over $\lambda$).  Through substitution, we find

$$ \frac{d F_{rad}}{d\tau_v} = 4\pi(\langle I \rangle - S). $$

**2) Multiplying by $\cos \theta$ and then integrating over all solid angles**

By multiplying first by $\cos \theta$, the integral on the left hand side now has the same form as the radiation pressure (Eqn. {eq}`rad_press_trans`).  Recall that $d\Omega = \sin \theta d\theta d\phi$, which is an odd function and the product of an odd and even function produces an odd function.  *An odd function integrated over symmetric limits is zero.*  First, the integrals are:

$$ \frac{d}{d\tau_v}\int I \cos^2 \theta d\Omega = \int I \cos\theta d\Omega - S \int \cos\theta d\Omega, $$

where the left hand side is replaced with the radiation pressure (with a factor of $c$; Eqn. {eq}`rad_press_trans`), the middle integral is the radiation flux, and the right most integral is zero.  Thus

```{math}
:label: rad_pressure_diff
\frac{d P_{rad}}{d\tau_v} = \frac{1}{c}F_{rad},
```

or 

$$ \frac{d P_{rad}}{dr} = -\frac{\kappa \rho}{c}F_{rad}, $$

in terms of a spherical coordinate system with its origin at the center fo the star.  As mention in [the radiation pressure gradient](https://saturnaxis.github.io/ModernAstro/Chapter_9/stellar-atmospheres.html#the-radiation-pressure-gradient), this result describes the net radiative flux driven by differences in radiation pressure (i.e., "photon wind").

For a plane-parallel atmosphere to be in equilibrium, the radiative flux must have the same value at every level, including its surface, or

\begin{align}  F_{rad} = {\rm constant} = F_{surf} = \sigma T_e^4. \end{align} 

This implies that the first of the above relations ($dF_{rad}/d\tau_v$) is zero and the mean intensity must equal the source function as

$$ \langle I \rangle = S. $$

Equation {eq}`rad_pressure_diff` can be integrated to find the radiation pressure as a function of the vertical optical depth:

```{math}
:label: rad_pressure
P_{rad} = \frac{\tau_v}{c}F_{rad} + C,
```

where $C$ is the constant of integration.

### The Eddington Approximation
The temperature as a function of the optical depth (i.e., temperature structure) can be determined if we knew how the radiation pressure varied with temperature in the general case.  Equation {eq}`rad_pressure` could be used if we also *assume* a description of the angular distribution of intensity.  An approximation was made by [Eddington](https://en.wikipedia.org/wiki/Arthur_Eddington), where the intensity of the radiation field is defined in the +$z$-direction $I_{\rm out}$ and another in the -$z$-direction $I_{\rm in}$.  These intensities vary with depth in the atmosphere and when $\tau_v = 0$, then $I_{\rm in} = 0$ (i.e., radiation doesn't flow inward at the top of the atmosphere).  The **Eddington approximation** gives the mean intensity, radiative flux, and radiation pressure by:

```{math}
:label: average_intens_Edd
\langle I \rangle = \frac{1}{2}(I_{\rm out} + I_{\rm in}), 
```

\begin{align}  F_{rad} = \pi(I_{\rm out} - I_{\rm in}), \end{align} 

and

```{math}
:label: rad_press_Edd
P_{rad}= \frac{2\pi}{3c}(I_{\rm out} + I_{\rm in}) = \frac{4\pi}{3c} \langle I \rangle .
```

Substituting Eqn. {eq}`rad_press_Edd` into Eqn. {eq}`rad_pressure`, we find that

```{math}
:label: rad_press_int
\frac{4\pi}{3c} \langle I \rangle = \frac{\tau_v}{c}F_{rad} + C, 
```

where the constant $C$ can be determined by evaluating at the top of the atmosphere ($\tau_v = 0$ and $I_{\rm in}=0$).  In this case, Eqn. {eq}`rad_press_Edd` becomes

$$ \frac{2\pi}{3c} I_{\rm out} = \frac{2}{3c} F_{rad} = C, $$

and substituting back into Eqn. {eq}`rad_press_int`, we find

\begin{align} \frac{4\pi}{3c} \langle I \rangle = \frac{\tau_v}{c}F_{rad} +  \frac{2}{3c} F_{rad} = \frac{F_{rad}}{c}\left(\tau_v + \frac{2}{3}\right).  \end{align} 

But we already know the radiative flux is a constant ($F_{rad} = \sigma T_e^4$) and thus, the mean intensity can be determined as a function of vertical optical depth by

```{math}
:label: mean_intensity_vod
\langle I \rangle  = \frac{3\sigma}{4\pi}T_e^4\left(\tau_v + \frac{2}{3}\right).
```

The final approximation of the temperature structure can be determined if the atmosphere is *assumed* to be in LTE ($S_\lambda = B_\lambda$).  Integrating the source function over all wavelengths produces

```{math}
:label: source_func
S = B = \langle I \rangle  = \frac{\sigma T^4}{\pi}.
```

Substituting Eqn. {eq}`source_func` back into Eqn. {eq}`mean_intensity_vod` results in the variation of the temperature with vertical optical depth in a plane-parallel gray atmosphere in LTE, assuming the Eddington approximation:

\begin{align}  T^4 = \frac{3}{4}T_e^4\left(\tau_v + \frac{2}{3}\right). \end{align} 

This relation reveals that $T=T_e$ at $\tau_v = 2/3$ and *not* at $\tau_v = 0$.  The "surface" of a star for which we measure an effective temperature $T_e$ is *not* at the top of the atmosphere, but deeper down where $\tau_v=2/3$.  The average point of origin for the photons comes from within the stellar atmosphere.  *When looking at a star, we see down to a vertical optical depth of $\tau_v \approx 2/3$, averaged over the disk of the star.*

### Limb Darkening Revisited
The previous discussion of [limb darkening](https://saturnaxis.github.io/ModernAstro/Chapter_9/stellar-atmospheres.html#limb-darkening) described limb darkening qualitatively.  Limb darkening is one measure of the how the source function varies with depth in a star's atmosphere.  Now we can examine this more closely starting with the transfer equation,

$$ \frac{dI_\lambda}{d\tau_\lambda} = I_\lambda - S_\lambda. $$

Multiplying both sides by $e^{-\tau_\lambda}$ and rearranging, we have

$$ \frac{dI_\lambda}{d\tau_\lambda} e^{-\tau_\lambda} - I_\lambda e^{-\tau_\lambda}= - S_\lambda e^{-\tau_\lambda}, $$

which is equivalent to

$$ \frac{d}{d\tau_\lambda}(I_\lambda e^{-\tau_\lambda}) = - S_\lambda e^{-\tau_\lambda}. $$

Integrating both sides with respect to $d\tau_\lambda$ gives

$$ \int_{t_{\lambda,o}}^0 \frac{d}{d\tau_\lambda}(I_\lambda e^{-\tau_\lambda}) = - \int_{t_{\lambda,o}}^0 S_\lambda e^{-\tau_\lambda} d\tau_\lambda, $$

where the left hand side is simply integrating the anti-derivative from the initial position of the ray at $\tau_{\lambda,o}$ ($I_\lambda = I_{\lambda,o}$) to the top of the atmosphere $\left(I_\lambda = I_\lambda(0) \right)$.  Evaluating the limits and rearranging produces

```{math}
:label: intensity_eq
I_\lambda(0) = I_{\lambda,o}e^{-\tau_{\lambda,o}} - \int_{\tau_{\lambda,o}}^0 S_\lambda e^{-\tau_\lambda} d\tau_\lambda.
```

Equation {eq}`intensity_eq` shows that the intensity we observe at the surface is the sum of two (positive) contributions.  The absorption term (first term on the RHS) describes the initial intensity, which is reduced along the path to the surface.  The emission (second term on the RHS) describes a modified emission that his attenuated by the absorption.

If we relax our assumptions (gray atmosphere, LTE, and Eddington approximation) but keep the plane-parallel atmosphere, then the problem of limb darkening is not very different, where each contribution is modified as a function of $\theta$.  This easily accomplished through the substitution $\tau_{\lambda,o} = \tau_{v,o} \sec \theta$ that uses the vertical optical depth $\tau_v$ to get

$$ I_\lambda(\theta) = I_{\lambda,o}e^{-\tau_{v,o} \sec \theta} - \int_{\tau_{v,o} \sec\theta}^0 S_\lambda\sec\theta\: e^{-\tau_{v\sec \theta}} d\tau_v. $$

```{figure-md} intensity-pp-fig
<img src="Figure_16.jpg" alt="plane parallel intensity"  width="450px">

Finding $I(\theta)$ for limb darkening in a plane-parallel geometry. Image credit: Carroll & Ostlie (2007).
```

Including the contributions from all layers of the atmosphere (or $\tau_{v,o}=\infty$), then the first term goes to zero leaving

```{math}
:label: intensity_int
I_\lambda(\theta) =  \int_0^\infty S_\lambda\sec\theta\: e^{-\tau_{v\sec \theta}} d\tau_v.
```

We need to know how the source function $S_\lambda$ depends on the optical depth to evaluate Eqn. {eq}`intensity_int`.  The exact form of $S_\lambda$ is not known, but a good guess will be to assume a linear dependence as

```{math}
:label: linear_src
S = a + b \tau_v,
```

where $a$ and $b$ are wavelength-dependent numbers.  Substituting Eqn. {eq}`linear_src` into Eqn. {eq}`intensity_int` and integrating, we get

```{math}
:label: solar_src_fcn 
I_\lambda(\theta) = a_\lambda + b_\lambda \sec \theta.
```

Careful measurements of the specific intensity of the Sun's disk allows for measurement of $a_\lambda$ and $b_\lambda$ for the Solar source function.

**Solar Limb Darkening**

Considering the Sun with a gray atmosphere, in LTE, and the Eddington approximation, we find that the source function (and mean intensity) is

$$ S = \frac{\sigma T^4}{\pi} = \frac{3\sigma}{4\pi} T_e^4 \left(\tau_v + \frac{2}{3}\right). $$

If we also assume that the source function is approximately linear with vertical optical depth (i.e., $S = a + b\tau_v$), then we can solve for the wavelength-dependent coefficients as

$$ a_\lambda = \frac{\sigma T_e^4}{2\pi}\quad {\rm and} \quad b_\lambda = \frac{3\sigma}{4\pi} T_e^4. $$

Using Eqn. {eq}`solar_src_fcn`, we have

\begin{align}  \frac{I(\theta)}{I(0)} = \frac{a_\lambda + b_\lambda \tau_v}{a_\lambda + b_\lambda} = \frac{2}{5} + \frac{3}{5}\cos \theta. \end{align} 


```{figure-md} Eddingtion-approx-fig
<img src="Figure_16.jpg" alt="Eddington approximation"  width="450px">

A theoretical Eddington approximation of solar limb darkening integrated over all wavelengths (curve) compared to observational data (dots) for the Sun. Image credit: Carroll & Ostlie (2007).
```

### **Problems**

>6\. If the temperature of a star's atmosphere is increasing outward, what type of spectral lines would you expect to find in the star's spectrum at those wavelengths where the opacity is greatest?
>
>7\. For a plane-parallel atmosphere, show that the Eddington approximation leads to expressions for the mean intensity, radiative flux, and radiation pressure given in [Eqns. {eq}`average_intens_Edd`-{eq}`rad_press_Edd`](https://saturnaxis.github.io/ModernAstro/Chapter_9/stellar-atmospheres.html#the-eddington-approximation). *Hint: Start with [Eqn. {eq}`mean_intensity`](https://saturnaxis.github.io/ModernAstro/Chapter_9/stellar-atmospheres.html#the-specific-and-mean-intensities) and apply boundary conditions.*
>
>8\. Complete Problem 28 (a-c) from Chapter 9 (Carroll & Ostlie (2007)).

## The Profiles of Spectral Lines

### Equivalent Widths
Absorption spectral lines are identified on graphs of the radiant flux $F_\lambda$ as a function of the wavelength $\lambda$.  The radiant flux can be normalized relative to the flux of the continuous spectrum $F_c$ outside the spectral line (i.e., $F_\lambda/F_c$).  Near the middle of the absorption (i.e., *core* of the spectral line), there is the central wavelength $\lambda_o$ and the *wings* extend outward, sweeping up to the continuum.  The depth $\delta$ of the line is simply the fractional decrease in radiant flux relative to the continuum (i.e., $\delta F_\lambda = 1 - F_\lambda/F_c$).  The strength of absorption is measured by its **equivalent width** $W$, which is a rectangle with height equal to unity and a specific width so that the area of the rectangle is equal to the total area removed from the continuum by the absorption.  Mathematically, this is

\begin{align}  W = \int 1- \frac{F_\lambda}{F_c} d\lambda = \int \delta F d\lambda, \end{align} 

where the integral is taken from one side of $\lambda_o$ to the other and is usually on the order of 0.01 nm.  Another measure of spectral line width uses a point at half of the depth at the core wavelength (i.e., $\delta F_{1/2} = 1/2 - F_{\lambda_o}/(2F_c)$) and moves to the left/right a $\Delta \lambda_{1/2} = \lambda_o \pm \lambda_{\delta F}$ until intersecting the absorption line so that $\delta F = \delta F_{1/2}$, or

$$ 1 - \frac{F_\lambda}{F_c} = \frac{1}{2}\left(1 - \frac{F_{\lambda_o}}{F_c} \right), $$

which is called the **full width at half-maximum** (FWHM).

```{figure-md} line-profile-fig
<img src="Figure_18.jpg" alt="line profile"  width="450px">

The profile of a typical spectral line. Image credit: Carroll & Ostlie (2007).
```

A spectral line is called **optically thin** when there is no wavelength at which the radiant flux is completely blocked (i.e., $F_\lambda/F_c = 0$).  The opacity $\kappa_\lambda$ is greatest at the core wavelength $\lambda_o$ and decreases moving into the wings.  The core wavelength is formed at higher (and cooler regions) of the stellar atmosphere, where the wings form at progressively deeper (and hotter) layers of the atmosphere, until it merges with the continuum producing region ($\tau_v = 2/3$).


### Processes That Broaden Spectral Lines
Three main processes are responsible for broadening spectral lines with mechanisms that produce a distinctive line shape or *line profile*.

1. **Natural broadening** occurs because the spectral lines cannot be sharp even for motionless, isolated atoms due to Heisenberg's uncertainty principle.  An electron occupies the orbital for the first excited state for an instant $\Delta t$ and the orbital's energy $E$ cannot have a precise value ($\Delta E\: \Delta t \approx \hbar$).  Electrons can transition from nearby orbitals to these "fuzzy" energy levels, which produces an uncertainty in the absorbed/emitted photons.  Using the wavelength form for the energy of a photon ($E = hc/\lambda$), we find that the uncertainty $\Delta \lambda$ is roughly <br> <br>

    \begin{align} \Delta \lambda \approx \frac{\lambda^2}{2\pi c}\left(\frac{1}{\Delta t_o} + \frac{1}{\Delta t_f} \right),  \end{align} 

    which depends on the lifetime of the electron in its initial state $\Delta t_o$ and final state $\Delta t_f$.
2. **Doppler broadening** occurs because the gas particles of the atmosphere have thermal motion that is described by the Maxwell-Boltzmann distribution function with a most probable speed of the atoms ($v_{mp} = \sqrt{2kT/m}$).  The wavelengths of the absorbed or emitted light are Doppler-shifted nonrelativistically according to $\Delta \lambda /\lambda = \pm |v_r|/c$.  The width of a Doppler broadened spectral line should be  <br> <br>

    $$ \Delta \lambda \approx \frac{2\lambda}{c}\sqrt{\frac{2kT}{m}}. $$

    ```{note}
    The radial velocity $v_r$ can be modified to include the effect of FWHM and the most probable turbulent velocity $v_{\rm turb}$ within the atmosphere to get 

    $$ (\Delta \lambda)_{1/2} = \frac{2\lambda}{c} \sqrt{\left(\frac{2kT}{m} + v_{\rm turb}^2 \right) \ln 2}, $$

    where the effect of turbulence is most important in the atmospheres of giant and supergiant stars.

    ```

    Other sources of Doppler broadening include stellar rotation, pulsation, and mass loss.  These phenomena can effect the shape and width of line profiles, but cannot be combined with Doppler broadening.

3. **Pressure (and collisional) broadening** occurs when the atomic orbitals are perturbed by a collision with a neutral atom or a close encounter with the electric field of an ion.  The effect of individual collisions is called *collisional broadening*, where the statistical effect of a large number of closely passing ions (with electric fields) is called *pressure broadening*.  In both cases the outcome depends on the average time between collisions or encounters. <br> <br>

    Precisely calculating the width and shape of a pressure-broadened line is complicated, where the atoms atoms involved can be the same or different elements.  The general shape is similar to natural broadening ($\Delta \lambda = \lambda^2/(\pi c \Delta t_o)$; natural broadening at FWHM), where the line profile resembles a radiation spectrum by an electric charge undergoing damped simple harmonic motion and thus termed as a *damping profile* (or *Lorentz profile*). <br> <br>

    An estimate of (collisional) pressure broadening with atoms of a single element can be made using $\Delta t_o$ as the average time between collisions.  The mean free path $\ell$ is approximately the product of the most probably speed with the average time between collision.  Using Eqn. {eq}`mean_free_path`, we find that <br> <br>

    $$ \Delta t_o \approx \frac{\ell}{v} = \frac{1}{n\sigma} \sqrt{\frac{m}{2kT}}, $$

    which depends on the atom mass $m$, the collisional cross section $\sigma$ and the number density $n$ of the atoms.  The spectral line width due to (collisional) pressure broadening is  <br> <br>

    \begin{align}  \Delta \lambda = \frac{\lambda^2}{\pi c \Delta t_o} \approx \frac{\lambda^2 n \sigma}{\pi c}\sqrt{\frac{2kT}{m}}. \end{align} 

    The spectral line width is proportional to $n$ and narrower lines for the more luminous giant and supergiant stars are due to the lower $n$ in their extended atmospheres (i.e., the Morgan-Keenan luminosity classes).  Pressure broadening is more significant in the denser atmospheres of main-sequence stars, where collisions occur more frequently.

- **What is the wavelength uncertainty for an electron in hydrogen ${\rm H}\alpha$ ($\lambda = 656.28$ nm), if its lifetime in the first and second excited states is about $\Delta t = 10^{-8}$ s?**

In [ ]:
import numpy as np
from scipy.constants import c #https://docs.scipy.org/doc/scipy/reference/constants.html
def delta_Natural(l,dt_o,dt_f):
    #l = lambda; wavelength (in nm)
    #dt_o = lifetime in the initial state (in sec)
    #dt_f = lifetime in the final state (in sec)
    return (l*1e-9)**2/(2.*np.pi*c)*(1./dt_o + 1./dt_f)

l_Halpha = 656.28 #nm
delta_l = delta_Natural(l_Halpha,1e-8,1e-8)
print("The wavelength uncertainty through natural broadening is approximately %1.2e m or %1.2e nm." % (delta_l,delta_l/1e-9))

- **What is the wavelength uncertainty due to Doppler broadening of the ${\rm H}\alpha$ line in the Sun's photosphere ($T = 5777$ K)?**
- **How many times more/less is this uncertainty compared to natural broadening?**

In [ ]:
from scipy.constants import m_p, c, k
def delta_Doppler(l,T):
    #l = lambda; wavelength (in nm)
    #T = temperature (in K)
    return 2.*(l*1e-9)/c*np.sqrt(2.*k*T/m_p)

delta_l_D = delta_Doppler(l_Halpha,5777)
print("The wavelength uncertainty through Doppler broadening is approximately %1.3e m or %1.5f nm" % (delta_l_D,delta_l_D/1e-9))
print("The Doppler broadening is %d times greater than natural broadening." % (delta_l_D/delta_l))

- **What is the wavelength uncertainty due to pressure broadening of the ${\rm H}\alpha$ line in the Sun's photosphere ($T = 5777$ K), where the hydrogen number density is about $1.5 \times 10^{23}\:{\rm m^{-3}}$?**
- **How many times more/less is this uncertainty compared to natural broadening?**

In [ ]:
def delta_Pressure(l,n,sigma,T):
    #l = lambda; wavelength (in nm)
    #n = number density of atoms (in m^{-3})
    #sigma = collisional cross section (in m^2)
    #T = temperature (in K)
    return (l*1e-9)**2*n*sigma/(np.pi*c)*np.sqrt(2*k*T/m_p)

def delta_Natural_FWHM(l,dt_o):
    #l = lambda; wavelength (in nm)
    #dt_o = average lifetime (in sec) for FWHM
    return (l*1e-9)**2/(np.pi*c*dt_o)

n_H = 1.5e23 #number density of H atoms
#from Sect 5.2.1
a_o = 5.29e-11 #m; radius of the Bohr atom
sigma_H = np.pi*(2*a_o)**2
l_Halpha = 656.28 #nm

delta_l_P = delta_Pressure(l_Halpha,n_H,sigma_H,5777)
delta_l_N = delta_Natural_FWHM(l_Halpha,1.9e-8)
print("The wavelength uncertainty through Pressure broadening is approximately %1.3e m or %1.3e nm" % (delta_l_P,delta_l_P/1e-9))
print("The Pressure broadening is %1.2f times that of natural broadening (Natural ~ Pressure)." % (delta_l_P/delta_l_N))


### The Voigt Profile
The total line profile (**Voight profile**) is due to the contributions of both the Doppler and damping (pressure) profiles.  The Doppler broadening dominates near the core wavelength, where the damping profile dominates in the wings at about $1.8\times$ the FWHM wavelength $(\Delta \lambda)_{1/2}$.  Some line profiles tend to have *Doppler cores* and *damping wings.*

```{figure-md} Doppler-profile-fig
<img src="Figure_19.jpg" alt="Doppler profile"  width="450px">

Schematic Doppler and damping line profiles scaled to have the same equivalent width. Image credit: Carroll & Ostlie (2007).
```

The simplest model used (**Schuster-Schwarzchild model** ) for calculating a line profile assume that the star's photosphere acts like a blackbody and atoms above the photosphere remove photons from the continuous spectrum.  This is inconsistent with the idea that photons originate at an optical depth $\tau_\lambda = 2/3$, but it is still a useful approximation.  To use the model, one must adopt values for the temperature, density, and composition for the region above the photosphere.  The temperature and density: 1) determine the importance of Doppler and pressure broadening and 2) are used in the Boltzmann and Saha equations.

Let $N$ be the number of atoms of a certain element per unit area lying above the photosphere, where $N$ represents a **column density** with units of ${\rm m^{-2}}$.  The temperature and density are used (in the Boltzmann and Saha equations) to calculate the excitation and ionization states that determine the number of absorbing atoms $N_a$ (i.e., number of atoms that have electrons in the proper orbital for absorption).  Our goal is to determine $N_a$ using the calculated and observed line profiles.  Quantum mechanics is used to determine the likelihood of transitions and the relative probabilities of an electron making a transition are given by *f-values* or *oscillator strengths*.  For hydrogen, $f=0.637$ for ${\rm H \alpha}$ and $f=0.119$ for ${\rm H\beta}$.  The oscillator strength may be numerically or experimentally determined, where they are defined so that the *f-values* for transitions (from the same initial orbital) add up to the number of electrons in the atom or ion.  *The oscillator strength is the product of $N_a$ and the f-value.* 


```{figure-md} Voigt-profile-fig
<img src="Figure_20.jpg" alt="Voigt profile"  width="700px">

Voigt profiles of the Ca II K line, where the shallowest line is produced by $N_a = 3.4 \times 10^{15}\:{\rm ions/m^2}$. The ions are ten times more abundant for each successively broader lines. Image credit: Carroll & Ostlie (2007).
```

### The Curve of Growth
The **curve of growth** helps determine the value of $N_a$ and the abundances of elements in stellar atmospheres.  Since the equivalent width $W$ varies with $N_a$, a curve of growth represents the logarithmic change in both quantities.  When an absorbing element is introduced it produces absorption lines that are linearly proportional to $N_a$, until a saturation point is reached and the core line is optically thick.  The wings are still optically thin and can deepen, so the curve of growth increases proportional to $\sqrt{\ln N_a}$.  For large numbers of absorbing atoms, the pressure broadening contributes to deepening the wings and the absorption width grows proportionally to $\sqrt{N_a}$.  Using the curve of growth and a measured equivalent width, we can obtain the number of absorbing atoms.

```{note}
To reduce the errors involved in a single spectral line, it is advantageous to locate (on a single curve of growth) the positions of equivalent widths of several lines formed by transitions from the same initial orbital.  However this in not applicable for all broad lines (e.g., H) and may lead to inaccurate results.  A logarithmic plot of the equivalent width per wavelength versus the *f*-value of the Number of atoms as a function of wavelength (normalized relative to 500 nm) provides this alternative view of the curve of growth (see Figure {numref}`{number}<solar-growth-fig>`).
```

```{figure-md} growth-curve-fig
<img src="Figure_21.jpg" alt="Curve of growth"  width="450px">

The curve of growth for the Ca II K line. Image credit: Carroll & Ostlie (2007).
```

**Absorption in the Sun's photosphere**

The Solar spectrum contains absorption lines of sodium at 330.238 and 588.997 nm.  Carroll and Ostlie (2007) assumed a particular temperature ($T = 5800$ K) and electron pressure ($P_e = 1\:{\rm N/m^2}$) for this example for the curve of growth.  **Table 1** provides equivalent widths, *f*-values, and values from the curve of growth as a function of wavelength.  *Note that the final column does not include the number of absorbing atoms per unit area $N_a$.*

<div align="center">

![Table 1](https://github.com/saturnaxis/ModernAstro/blob/main/Chapter_9/Table_1.jpg?raw=true?)<br>
**Table 1**: Data for Solar sodium lines. ([Aller (1991)](https://ui.adsabs.harvard.edu/abs/1991asn..book.....A/abstract))

</div>

Sodium lines are produced during upward electron transitions from the ground state of the Na I atom and thus, these lines have the same value of $N_a$ above the continuum-forming layer of the photosphere.  The value for $N_a$ can be determined using the values of $\log_{10}(W/\lambda)$ from Table 1 to locate the respective values of $\log_{10}(fN_a(\lambda/500\:{\rm nm}))$ in Figure {numref}`{number}<solar-growth-fig>` (i.e., follow the arrows).

```{figure-md} solar-growth-fig
<img src="Figure_22.jpg" alt="Curve of growth Solar"  width="600px">

The general curve of growth for the Sun. Image credit: Carroll & Ostlie (2007); data taken from [Aller (1991)](https://ui.adsabs.harvard.edu/abs/1991asn..book.....A/abstract).
```

Using Fig. {numref}`{number}<solar-growth-fig>`, we find values of 17.15 and 18.83 for the 330.238 and 588.997 nm lines, respectively.  Therefore, the respective $N_a$ values can be determined from the *difference of values* from the curve of growth (Fig. {numref}`{number}<solar-growth-fig>`) and the tabulated values (Table 1).  It follows

\begin{align*}
\log_{10}\:N_a &= \log_{10}\left(\frac{fN_a\lambda}{500\:{\rm nm}}\right) - \log_{10}\left(\frac{f\lambda}{500\:{\rm nm}}\right)  \\
&= 17.15 - (-1.85) = 19.00 \quad {\rm for\:the\:330.238\:nm\:line}
\end{align*}

and

\begin{align*}
\log_{10}\:N_a &= \log_{10}\left(\frac{fN_a\lambda}{500\:{\rm nm}}\right) - \log_{10}\left(\frac{f\lambda}{500\:{\rm nm}}\right)  \\
&= 18.83 - (-0.12) = 18.95 \quad {\rm for\:the\:588.997\:nm\:line.}
\end{align*}

The average value of $\log_{10}\:N_a$ is 18.96 and thus, there are ~$10^{19}$ ground state Na I atoms per ${\rm m^2}$ of the photosphere.  The [Boltzmann and Saha equations](https://saturnaxis.github.io/ModernAstro/Chapter_8/classification-of-stellar-spectra.html?highlight=saha#combining-the-boltzmann-and-saha-equations) are needed to find the total number of sodium atoms.  The energy of each emitted photon can be calculated using $E_\lambda = hc/\lambda$ and used as the difference in energy states ($E_b - E_a$) in the Boltzmann equation, so that

\begin{align*}
e^{-(E_b-E_a)/(kT)} &= e^{-hc/(\lambda kT)} \\
&= 5.47 \times 10^{-4} \quad {\rm for\:the\:330.238\:nm\:line} \\
& = 1.48 \times 10^{-2} \quad {\rm for\:the\:588.997\:nm\:line},
\end{align*}

and unsurprisingly, nearly all of the Na I atoms are in the ground state.  

In [ ]:
from scipy.constants import h,c,k
import numpy as np

def Boltzmann_eq(l,T):
    #l = lambda; wavelength (in nm)
    #T = temperature (in K)
    return np.exp(-(h*c)/((l*1e-9)*k*T))

e_330 = Boltzmann_eq(330.238,5800)
e_588 = Boltzmann_eq(588.997,5800)

print("The Boltzmann exponential term for the 330.238 line is %1.2e." % e_330)
print("The Boltzmann exponential term for the 588.997 line is %1.2e." % e_588)

The total number of Na atoms per unit area in all ionization stages requires the Saha equation.  Using the curve of growth, we estimated the number of Na I atoms in the ground state ($N_I = 10^{19}$), then the number of singly ionize atoms $N_{II}$ is determined by

$$ \frac{N_{II}}{N_I} = \frac{2kTZ_{II}}{P_e Z_I}\left( \frac{2\pi m_e kT}{h^2}\right)^{3/2} e^{-\chi_I/(kT)}. $$

Using $Z_{II} = 1.0$ and $Z_{I} = 2.4$ for the partition functions and the ionization energy ($\chi_I = 5.14$ eV) of Na I leads to $N_{II}/N_{I} = 2.43 \times 10^3$.  There are ~2220 Na II atoms for every Na I atom in the Sun's photosphere and the total number of Na atoms per unit area above the photosphere is about $2.22 \times 10^{22}\:{\rm m^{-2}}$. 

In [ ]:
from scipy.constants import h,k,m_e,e
import numpy as np

def Saha_Eq(Z_I,Z_II,T,chi_I):
    #Z_I = neutral state partition function 
    #Z_II = singly ionized partition function
    #T = temperature (in K)
    #chi_I = ionization energy from the ground state (in eV)
    return (2*k*T*Z_II)/(P_e*Z_I)*(2*np.pi*m_e*k*T/h**2)**1.5*np.exp(-(chi_I*e)/(k*T))

P_e = 1. #1 N/m^2 assumed by Carroll and Ostlie (2007)
NII_NI = Saha_Eq(2.4,1.0,5800,5.14)
tot_Na = NII_NI*10**(18.96)

print("The ratio N_II/N_I for Na is %1.2e." % NII_NI)
print("The total number of Na atoms per unit area above the photosphere is %1.2e." % tot_Na)

m_Na = 3.82e-26 #kg; mass of sodium
print("The density of Na above each m^2 of the photosphere is %1.2e kg/m^2" % (m_Na*tot_Na))

The number of absorbing atoms can be determined by comparing the measured equivalent widths for different absorption lines with a theoretical curve of growth.  A similar analysis can be performed for lines originating from atoms (or ions) in different initial states, where the Boltzmann equation is applied to the relative numbers of atoms (or ions) to calculate the excitation temperature.  It also possible to use the Saha equation to find either the electron pressure of the ionization temperature.

### Computer Modeling of Stellar Atmospheres

The ultimate refinement in analyzing stellar atmospheres comes from the construction of a *model atmosphere* on a computer.  Each of the parameters that we use (temperature, pressure, and density) depend on each other in complex ways.  Each atmospheric layer is involved in the formation of line profiles that contribute to the observed stellar spectrum.  The equations of hydrostatic equilibrium, thermodynamics, statistical and quantum mechanics, and energy transport (by radiation and convection) are combined with extensive libraries of opacities to calculate the temperature, pressure, and density as a function of depth below the surface.

<div align="center">

![Table 2](https://github.com/saturnaxis/ModernAstro/blob/main/Chapter_9/Table_2.jpg?raw=true?)<br>
**Table 2**: Elemental abundance in the Solar photosphere.  The relative abundance refers to the abundance relative to H ($\log_{10}\:N_H=12.00$) ([Grevesse & Sauval (1998)](https://ui.adsabs.harvard.edu/abs/1998SSRv...85..161G/abstract); Carroll & Ostlie (2007))

</div>

The models provide insight into measurable fundamental properties (surface temperature and surface gravity) of a star.  After extensive minimization (or "fine-tuning") relative to the observations, astronomers can finally claim to understand the coded information emanating as light from a star.  This basic procedure led astronomers to an understanding of the elemental abundances of the Sun and other stars.  Knowledge of the basic ingredients of the universe provides observational tests and constraints for some of the most fundamental theories in astronomy:

- the nucleosynthesis of light elements as a result of stellar evolution,
- the production of heavier elements by supernovae, and
- the Big Bang that produced the primordial H and He that started it all.

## Supplemental Links
- [Powerpoint slides](http://www.ifa.hawaii.edu/users/kud/teaching_12/3_Radiative_transfer.pdf) from Rolf Kudritzki at the University of Hawaii's Institute for Astronomy (IfA).
- [Integrating the Planck function](http://people.tamu.edu/~kevinkrisciunas/planck.pdf) from Kevin Krisciunas at Texas A&M University.
- Contour integral for [Planck function](https://math.stackexchange.com/questions/99843/contour-integral-for-x3-ex-1).
- [Notes on Optical Depth](http://spiff.rit.edu/classes/phys440/lectures/optd/optd.html) from Michael Richmond at Rochester Institute of Technology.
- [Notes on Eddington Approximation](http://www.astro.wisc.edu/~townsend/resource/teaching/astro-310-F08/13-gray-eddington.pdf) from Richard Townsend at University of Wisconsin-Madison.